discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+
+A fediverse server MUST make sure it only shares content that it is
+actually allowed to share. It MUST NOT share any content that is not
+public. In the case of account and post data it MUST make sure to only
+share content where the creator has opted in to discovery.
+
+The FASP in turn MUST also ensure that creators have opted in to
+discovery before storing and indexing content.
+
+Both parties MUST use the mechanism described in
+[FEP-5feb](https://codeberg.org/fediverse/fep/src/branch/main/fep/5feb/fep-5feb.md)
+to determine if a creator has opted in to discovery.
+
+For account data, both parties MUST ensure the `discoverable` flag is
+set to `true`.
+
+## Subscribing To, Requesting And Receiving Content
+
+### Managing Subscriptions (FASP => Fediverse Server)
+
+In order to subscribe to new content, FASPs can subscribe to events
+by making a `POST` call to the `/data_sharing/v0/event_subscriptions`
+endpoint on the fediverse server.
+
+Example call:
+
+```http
+POST /data_sharing/v0/event_subscriptions
+```
+
+The request body MUST contain a JSON object defining what events to
+subscribe to. This object MUST contain the keys `objectType` and
+`eventType`. It MAY contain the keys `maxBatchSize` and/or
+`threshold`. The keys are defined as follows:
+
+* `category`: One of `content`, `account`. This is the category of
+  objects the FASP is interested in.
+* `subscriptionType`: One of `lifecycle`, `trends`. `lifecycle`
+  means the FASP would like to be notified when new objects of the given
+  type are created, when objects are updated and when objects are
+  deleted. `trends` means that the FASP would like to be notified of
+  objects that had recent interactions. `trends` is only applicable if
+  the `category` given is `content`.
+* `maxBatchSize`: The maximum number of events the FASP would like to
+  receive in a single request. This is optional.
+* `threshold`: If `eventType` is `trending` then this object can be used to
+  further specify when the event should be reported. Valid keys for this
+  object are:
+  * `timeframe`: Number of minutes in which interactions should fire an
+    event, defaults to `15`
+  * `shares`: Number of shares in the given timeframe, defaults to `3`
+  * `likes`: Number of likes in the given timeframe, defaults to `3`
+  * `replies`: Number of replies in the given timeframe, defaults to `3`
+
+Example object:
+
+```json
+{
+  "category": "content",
+  "subscriptionType": "trends",
+  "maxBatchSize": 10,
+  "threshold": {
+    "timeframe": 15,
+    "shares": 3
+  }
+}
+```
+
+The fediverse server MUST validate the request. If it is invalid it MUST
+return an HTTP status code `422` (Unprocessable Content).
+
+If it is valid the response from the fediverse server MUST include an
+HTTP status code `201` (Created) and a JSON object including the `id` of
+the generated subscription. The latter can be used to unsubscribe later.
+
+Example response object:
+
+```json
+{
+  "subscriptionId": 3446
```

Comment:
I'd generally suggest using an object here:
```suggestion
  "subscription": {
    "id": "3446"
  }
```

(I'm thinking subscription IDs may be bigints, so should probably be strings.

Reply:
That way if we need to add more properties to the response later, then it's much easier to do so.

Reply:
Good idea, thanks!